### PR TITLE
Fix/sw src

### PR DESCRIPTION
**Pull request by rschristian**
**What kind of change does this PR introduce?**

Bug fix

**Did you add tests for your changes?**

No

**Summary**

With ESM disabled, `InjectManifest` used `join(src, sw.js)` to get the service worker path. If this did not exist (if the user [re]moved the file), the build would error out.

We already had a system set up to catch this and fall back to the one provided from `preact-cli`, just weren't using it on non-esm builds.

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -127,7 +127,7 @@
     "stack-trace": "0.0.10",
     "style-loader": "^2.0.0",
     "terser-webpack-plugin": "^4.2.3",
-    "typescript": "^4.2.4",
+    "typescript": "~4.2.4",
```
Comment (rschristian) [inline packages/cli/package.json:130]: This is completely unrelated so I could move into a separate PR if desired.

From what I understand, Yarn needs to manually patch some dependencies like TS on every release. TS 4.5 just came out, which apparently caused some issues in the PnP pipeline.

Shouldn't be an issue to just restrict this to a patch/TS minor

Comment (developit) [reply packages/cli/package.json:130]: eh, if it makes CI pass leave it in.

**Review by developit — APPROVED** [review]
